packages/uicore/src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -103,15 +104,16 @@ export const ModalDialog: FC<ModalDialogProps> = ({
 
   useEffect(() => bodyObserverRef.current?.disconnect, [])
 
-  const onOpening: IDialogProps['onOpening'] = arg => {
-    initScrollShadows()
-    dialogProps.onOpening?.(arg)
-  }
-
-  const onClose: IDialogProps['onClose'] = arg => {
-    bodyObserverRef.current?.disconnect()
-    dialogProps.onClose?.(arg)
-  }
+  const bodyRefCallback = useCallback(
+    (el: HTMLDivElement | null) => {
+      if (!el) {
+        return bodyObserverRef.current?.disconnect()
```

Comment:
We missed a null check here : disconnect?.(), this is causing a jest for one of modal to break. Fixing this.